pyparsing/core.py

```diff
@@ -2947,18 +2951,36 @@ def __init__(
                 "Regex may only be constructed with a string or a compiled RE object"
             )
 
-        self.re_match = self.re.match
-
         self.errmsg = "Expected " + self.name
         self.mayIndexError = False
-        self.mayReturnEmpty = self.re_match("") is not None
         self.asGroupList = asGroupList
         self.asMatch = asMatch
         if self.asGroupList:
             self.parseImpl = self.parseImplAsGroupList
         if self.asMatch:
             self.parseImpl = self.parseImplAsMatch
 
+
+    @cached_property
+    def re(self):
+        if self._re:
```

Comment:
Wouldn't it be possible to always assign to re and thus remove the extra attribute for a given pattern

Reply:
no, see `__init__`

Reply:
@asottile my understanding is that a cached property can be assigned a value using the same attribute name, as its using the __dict__ as cache

that way the assigment should suffice

Reply:
yes, but `self._re` is used separately